### PR TITLE
Leaderboard project table

### PR DESCRIPTION
**Pull request by swellander**
## Description
Adds the project table for the leaderboard page. The table sorts all projects by their % movement based on the selected metric.

[Screencast from 2024-07-17 15-12-14.webm](https://github.com/user-attachments/assets/3c2dbfb9-ed88-4c35-b028-e6268d936fe5)

### CODE REVIEW

File: packages/nextjs/components/onchain-impact-dashboard/LeaderboardTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  selectedMetricName: string;
+}
+
+type ColumnHeader = keyof IProjectMovement;
+
+const formatMovement = (movement: number) => {
+  if (movement >= 0) {
+    return `+ ${movement}%`;
+  }
+  return `- ${movement * -1}%`;
```
Comment (escottalexander) [inline packages/nextjs/components/onchain-impact-dashboard/LeaderboardTable/index.tsx:25]: Seems like this will show double negative? "- -11%"

**Review by escottalexander — CHANGES_REQUESTED** [review]
This looks great! Left one small comment. Great work here! The Skeleton is really good.

File: packages/nextjs/components/onchain-impact-dashboard/LeaderboardTable/index.tsx (hunk 1 above)
Comment (swellander) [reply packages/nextjs/components/onchain-impact-dashboard/LeaderboardTable/index.tsx:25]: This is just to make sure the padding of the `-` symbol is the same as the `+` symbol. If the movement is positive, we prefix the number with a `+ `. If the movement is negative, we flip it to positive to remove the automatic `-` symbol, and then prefix it with our own `- ` with the correct spacing.

**Review by Franpastoragusti — APPROVED** [review]

File: packages/nextjs/components/onchain-impact-dashboard/LeaderboardTable/index.tsx (hunk 1 above)
Comment (escottalexander) [reply packages/nextjs/components/onchain-impact-dashboard/LeaderboardTable/index.tsx:25]: That makes sense! Thanks!

**Review by escottalexander — APPROVED** [review]
Nice work!

**Pull request merged**